drivers/flash/nrf_qspi_nor.c

```diff
@@ -432,7 +432,7 @@ static int qspi_erase(const struct device *dev, uint32_t addr, uint32_t size)
 	qspi_trans_lock(dev);
 	rv = qspi_nor_write_protection_set(dev, false);
 	qspi_lock(dev);
-	while ((rv == 0) && (size > 0)) {
+	while (size > 0) {
```

Comment:
Previously the loop was not executed if `qspi_nor_write_protection_set()` was not successful. I'm not convinced that this behavior should be changed.

Reply:
Fixed, the loop will be jumped over.